### PR TITLE
optimize CI

### PR DESCRIPTION
**Pull request by Lotterleben**
~opening as PR to trigger ci builds.~

closes https://github.com/knurling-rs/flip-link/issues/7 ~once it's done~

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -61,16 +71,18 @@ jobs:
           target: ${{ env.NRF_TARGET }}
       - name: Install dependencies
         run: |
+          pwd
```
Comment (Lotterleben) [inline .github/workflows/ci.yml:74]: note to self: dont forget to rm this

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -61,16 +76,22 @@ jobs:
...
-          cargo generate --git https://github.com/knurling-rs/app-template --branch main --name ${{ env.APPNAME }}
-          cd ${{ env.APPNAME }}
-          git apply ../.github/workflows/0001-configure-test-flip-link-app-for-nrf52840.patch
+          cd ${{ env.APP_NAME }}
+          git apply ../${{ env.REPO_NAME }}/.github/workflows/0001-configure-test-flip-link-app-for-nrf52840.patch
       - name: Build App
         run: |
-          cd ${{ env.APPNAME }}
+          cd ${{ env.APP_NAME }}
+          cat Cargo.toml
```
Comment (japaric) [inline .github/workflows/ci.yml:96]: leftover debug statement?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -61,16 +76,22 @@ jobs:
           target: ${{ env.NRF_TARGET }}
       - name: Install dependencies
         run: |
-          sudo apt-get update && sudo apt-get install libssl-dev
+          if [ "$RUNNER_OS" == "Linux" ]; then
+            sudo apt-get update && sudo apt-get install libssl-dev
```
Comment (japaric) [inline .github/workflows/ci.yml:80]: do we still need to install openssl after removing the `cargo-generate` installation?

**Review by japaric — APPROVED** [review]
Looks good to me! I think we may be able to remove the openssl install (not sure)

File: .github/workflows/ci.yml (hunk 3 above)
Comment (Lotterleben) [reply .github/workflows/ci.yml]: you're right, we don't– removed :)

**Pull request merged**